### ISSUE TITLE
Dynamic Area Target Creation During Runtime

### ISSUE THREAD

**Issue by LucaBraun** [PTCInc/vuforia-engine#19]
**Summarize your feature feedback**
We want our future customers to be able to upload a 3d scan of their environment to our app and be able to use it as an Area Target. This would require some kind of SDK feature that reflects the capabilities of the Area Target Generator. As far as we can tell from the forums this used to be supported by the SDK but has been deprecated now.

**Describe your feature**
Some kind of Library, SDK feature, or API to let clients convert supported file types to Area Targets in runtime of our Vuforia app.

**Use-case description**
Since our application should work in different and for us at the point of development unknown environments we wanted to be able to upload 3d models and use those as area targets during runtime

**Business impact**
It would make it possible for us to use Vuforia as our AR solution.

**Workaround**
There sadly isn't a workaround as far as we know.

**Additional context**
We have to use area targets since our application is used for the calibration of sensors that are placed in the environment. It is necessary for us to get to know their position within the environment. This is only possible using Area Targets as far as we know.

**Comment by ptc-pscheper** [comment]
Hi there, you're able to make this work by utilizing the Area Target Capture API, but this would require you to capture the targets on device. If you're using Matterport scans or .E57 data from supported scanners you're right, this isn't possible as of today, but I've relayed your feedback request to the team! Thank you for sharing your thoughts.

**Comment by davidkanaga** [comment]
### Our Case:

We want to dynamically load Area Target Dataset at runtime because our app size is increasing drastically whenever we are adding new dataset. We are scanning entire mountains and it's sizes are too big. Need a solution ASAP.

### Answer from ChatGPT:

To clarify, in Vuforia 10.15.4, there isn't a straightforward way to dynamically load an Area Target dataset at runtime. The Area Target feature in Vuforia is primarily designed for static targets that are defined during the development phase rather than dynamically loaded at runtime.

**Comment by ptc-pscheper** [comment]
Can you confirm that your challenge is the size of the app or is it the inability to automate your backend workflows to generate Area Targets?

**Comment by davidkanaga** [comment]
> Can you confirm that your challenge is the size of the app or is it the inability to automate your backend workflows to generate Area Targets?

We can able to generate Area Targets there is no problem in it. But adding the dataset on the development phase itself increasing the size of the app. So we need a solution by dynamically loading the dataset at runtime.

**Comment by davidkanaga** [comment]
> So you'd need an ability to load parts of the Area Target during runtime so you don't have to load the huge Area Targets you spoke of entirely?

Yup that's correct, we would like to download the area targets from our cloud storage then attach it to the app in runtime. For example how we use asset bundles.

Our App link : [https://www.red-point.com.au/](url)

**Comment by ptc-emaggio** [comment]
Hi @LucaBraun and @davidkanaga we have just released our new Cloud Area Target API where targets are stored in our cloud infrastructure and streamed as-needed to the devices.

Here is the relevant [documentation](https://developer.vuforia.com/library/vuforia-engine/web-api/cloud-area-targets-web-api/).

**Comment by ptc-emaggio** [comment]
Closing this feature request as it is now possible to create a target a runtime by using the [Cloud Area Target API](https://developer.vuforia.com/library/vuforia-engine/web-api/cloud-area-targets-web-api/)
